plugins/weather.py

```diff
@@ -136,72 +147,71 @@ def weather(text, reply, db, nick, notice_doc):
-        "wind_kph": current_observation['wind_kph'],
-        "wind_mph": current_observation['wind_mph'],
-        "wind_direction": current_observation['wind_dir'],
-        "today_conditions": forecast_today['conditions'],
-        "today_high_f": forecast_today_high['fahrenheit'],
-        "today_high_c": forecast_today_high['celsius'],
-        "today_low_f": forecast_today_low['fahrenheit'],
-        "today_low_c": forecast_today_low['celsius'],
-        "tomorrow_conditions": forecast_tomorrow['conditions'],
-        "tomorrow_high_f": forecast_tomorrow_high['fahrenheit'],
-        "tomorrow_high_c": forecast_tomorrow_high['celsius'],
-        "tomorrow_low_f": forecast_tomorrow_low['fahrenheit'],
-        "tomorrow_low_c": forecast_tomorrow_low['celsius'],
-    }
-
-    # Get the more accurate URL if available, if not, get the generic one.
-    ob_url = current_observation['ob_url']
-    if "?query=," in ob_url:
-        url = current_observation['forecast_url']
-    else:
-        url = ob_url
-
-    weather_data['url'] = web.try_shorten(url)
-
-    reply("{place} - \x02Current:\x02 {conditions}, {temp_f}F/{temp_c}C, {humidity}, "
-          "Wind: {wind_mph}MPH/{wind_kph}KPH {wind_direction}, \x02Today:\x02 {today_conditions}, "
-          "High: {today_high_f}F/{today_high_c}C, Low: {today_low_f}F/{today_low_c}C. "
-          "\x02Tomorrow:\x02 {tomorrow_conditions}, High: {tomorrow_high_f}F/{tomorrow_high_c}C, "
-          "Low: {tomorrow_low_f}F/{tomorrow_low_c}C - {url}".format_map(weather_data))
+    fio = ForecastIO(
+        ds_key, units=ForecastIO.UNITS_US,
+        latitude=location_data['lat'], longitude=location_data['lng']
+    )
+
+    daily_conditions = fio.get_daily()['data']
+    current = fio.get_currently()
+    today, tomorrow = daily_conditions[:2]
+    current['name'] = 'Current'
+    today['name'] = 'Today'
+    tomorrow['name'] = 'Tomorrow'
+
+    for forecast in (current, today, tomorrow):
+        wind_speed = forecast['windSpeed']
+        forecast.update(
+            wind_direction=bearing_to_card(forecast['windBearing']),
+            wind_speed_mph=wind_speed,
+            wind_speed_kph=mph_to_kph(wind_speed),
+            summary=forecast['summary'].rstrip('.'),
+        )
+
+    current.update(
+        temp_f=current['temperature'],
+        temp_c=convert_f2c(current['temperature'])
+    )
+
+    for day_forecast in (today, tomorrow):
+        high = day_forecast['temperatureHigh']
+        low = day_forecast['temperatureLow']
+        day_forecast.update(
+            temp_high_f=high,
+            temp_high_c=convert_f2c(high),
+            temp_low_f=low,
+            temp_low_c=convert_f2c(low),
+        )
+
+    current_str = "\x02{name}\x02: {summary}, {temp_f:.2f}F/{temp_c:.2f}C " \
+                  "{humidity:.0%}, " \
+                  "Wind: {wind_speed_mph:.2f}MPH/{wind_speed_kph:.2f}KPH " \
+                  "{wind_direction}"
+    day_str = "\x02{name}\x02: {summary}, " \
```

Comment:
Do we really need temps to two decimal points?

Reply:
Follows the existing format

Reply:
There really isn't any reason to not have it, following existing or not. The world is not going to explode if you make it a tiny bit less verbose